central/debug/service/service.go

```diff
@@ -214,6 +234,31 @@ func (s *serviceImpl) StreamAuthzTraces(_ *v1.Empty, stream v1.DebugService_Stre
 	}
 }
 
+func getJSONToZipWithLogging(ctx context.Context, zipWriter *zip.Writer, fileName string, fetchData func(ctx context.Context) (interface{}, error)) {
+	jsonObj, errFetchData := fetchData(ctx)
+	if errFetchData != nil {
+		log.Error(errFetchData)
+
+		return
+	}
+
+	if errAddToZip := addJSONToZip(zipWriter, fileName, jsonObj); errAddToZip != nil {
+		log.Error(errAddToZip)
+	}
+}
+
+func addJSONToZip(zipWriter *zip.Writer, fileName string, jsonObj interface{}) error {
+	w, err := zipWriter.Create(fileName)
+	if err != nil {
+		return errors.Wrapf(err, "unable to create zip file: %q", fileName)
```

Comment:
```suggestion
		return errors.Wrapf(err, "unable to create zip file %q", fileName)
```
This way it will render as `unable to create zip file 'blah.json': all sold`.